oak_runtime/src/lib.rs

```diff
@@ -1164,6 +1165,26 @@ impl Runtime {
         config: &NodeConfiguration,
         label: &Label,
         initial_handle: oak_abi::Handle,
+    ) -> Result<(), OakStatus> {
+        // This only creates a Node instance, but does not start it.
+        let instance = self.node_factory.create_node(config).map_err(|err| {
+            warn!("could not create node: {:?}", err);
+            OakStatus::ErrInvalidArgs
+        })?;
+
+        // Register the instance within the `Runtime`.
+        self.register_node_instance(node_id, instance, config, label, initial_handle)
+    }
+
+    /// Registers the given [`Node`] instance within the [`Runtime`]. The registration fails if the
+    /// labels violate the IFC rules.
```

Comment:
Now that we have this method, do we still need `node_configure_instance`? I think that was just there to help with testing too, but I think this new method is overall a much better abstraction, so we should be able to change the previous test to just use this, and delete / inline the other one?

Reply:
Got it, sounds good.

BTW I was hoping that the "implicit" initial node can also be removed now? Or is it still needed for anything? I guess it is used in order to send the config map to the initial node, but I think now it should be possible to use the new method you just added to create a proper node to do that (without having to go through NodeFactory). WDYT? Could you create an issue if you think it makes sense?

https://github.com/project-oak/oak/blob/27f580699a84e78d7b36c4883ef77b590950c636/oak_runtime/src/proxy.rs#L78-L84